File: src/main/java/me/onetwoeight/clearchat/listener/ChatListener.java
package me.onetwoeight.clearchat.listener;

import me.onetwoeight.clearchat.Main;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.jetbrains.annotations.NotNull;

import static java.util.Objects.requireNonNull;
import static org.bukkit.Bukkit.broadcastMessage;
import static org.bukkit.Bukkit.getPlayer;
import static org.bukkit.ChatColor.*;

/**
 * @author onetwoeight
 * @since 3/27/2022
 */
public final class ChatListener implements CommandExecutor {

    private final Main plugin;

    public ChatListener(final Main plugin) {
        this.plugin = plugin;
    }

    @Override
    public boolean onCommand(@NotNull final CommandSender sender, @NotNull final Command command, @NotNull final String label, @NotNull final String[] args) {
        final String player = "cc.player";
        final String global = "cc.global";
        final String send = "%sender%";
        final String prefix = "Prefix";
        final String getPlayer = "Expression 'getPlayer(args[0])' must not be null";
        final String getString = "Expression 'plugin.getConfig().getString(\"String Path\")' must not be null";
        if (sender.hasPermission(global) && args.length == 0) {
            for (int i = 0; i < 1000; i++) {
                broadcastMessage("");
            }
            broadcastMessage(translateAlternateColorCodes('&', plugin.getConfig().getString(prefix) + requireNonNull(plugin.getConfig().getString("Global"), getString).replace(send, sender.getName())));
        } else if (sender.hasPermission(player) && args.length == 1) {
            for (int i = 0; i < 1000; i++) {
                if (getPlayer(args[0]) != null && requireNonNull(getPlayer(args[0]), getPlayer).isOnline()) {
                    requireNonNull(getPlayer(args[0]), getPlayer).sendMessage("");
                } else {
                    sender.sendMessage(RED + "Could not find specified player" + RESET);
                    break;
                }
            }
            if (getPlayer(args[0]) != null && requireNonNull(getPlayer(args[0]), getPlayer).isOnline()) {
                requireNonNull(getPlayer(args[0]), getPlayer).sendMessage(translateAlternateColorCodes('&', plugin.getConfig().getString(prefix) + requireNonNull(plugin.getConfig().getString("Player"), getString).replace(send, sender.getName())));
                sender.sendMessage(translateAlternateColorCodes('&', plugin.getConfig().getString(prefix) + requireNonNull(plugin.getConfig().getString("Success"), getString).replace("%person%", args[0])));
            }
        } else if (!sender.hasPermission(global) && args.length == 0 || !sender.hasPermission(player) && args.length > 0) {
            sender.sendMessage(translateAlternateColorCodes('&', requireNonNull(plugin.getConfig().getString("NoPermission"), getString).replace(send, sender.getName())));
        } else if (sender.hasPermission(player) && args.length > 1) {
            sender.sendMessage(RED + "Please refrain from using 2 or more args" + RESET);
        }
        return true;
    }
}

File: src/main/kotlin/me/onetwoeight/clearchat/listeners/ChatListener.kt
package me.onetwoeight.clearchat.listeners

import me.onetwoeight.clearchat.Main
import org.bukkit.Bukkit
import org.bukkit.ChatColor
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender

/**
 * @author onetwoeight
 * @since 4/14/2022
 */
internal class ChatListener(private val plugin: Main) : CommandExecutor {

    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
        val prefix = "Prefix"
        val send = "%sender%"
        val global = "cc.global"
        val player = "cc.player"
        if (sender.hasPermission(global) && args.isEmpty()) {
            for (i in 0..999) {
                Bukkit.broadcastMessage("")
            }
            Bukkit.broadcastMessage(
                ChatColor.translateAlternateColorCodes(
                    '&',
                    plugin.config.getString(prefix) + plugin.config.getString("Global")
                ).replace(send, sender.name)
            )
        } else if (sender.hasPermission(player) && args.size == 1) {
            for (i in 0..999) {
                if (Bukkit.getPlayer(args[0])?.isOnline == true) {
                    Bukkit.getPlayer(args[0])?.sendMessage("")
                } else {
                    sender.sendMessage("${ChatColor.RED}Could not find specified player${ChatColor.RESET}")
                    break
                }
            }
            if (Bukkit.getPlayer(args[0])?.isOnline == true) {
                Bukkit.getPlayer(args[0])?.sendMessage(
                    ChatColor.translateAlternateColorCodes(
                        '&',
                        plugin.config.getString(prefix) + plugin.config.getString("Player")?.replace(
                            send,
                            sender.name
                        )
                    )
                )
                sender.sendMessage(
                    ChatColor.translateAlternateColorCodes(
                        '&',
                        plugin.config.getString(prefix) + plugin.config.getString("Success")?.replace(
                            "%person%",
                            args[0]
                        )
                    )
                )
            }
        } else if (!sender.hasPermission(global) && args.isEmpty() || !sender.hasPermission(player) && args.isNotEmpty()) {
            sender.sendMessage(
                plugin.config.getString("NoPermission")?.let {
                    ChatColor.translateAlternateColorCodes(
                        '&', it.replace(send, sender.name)
                    )
                }
            )
        } else if (sender.hasPermission(player) && args.size > 1) {
            sender.sendMessage("${ChatColor.RED}Please refrain from using 2 or more args${ChatColor.RESET}")
        }
        return true
    }
}